test/board_test.exs

```diff
@@ -51,35 +51,61 @@ defmodule BoardTest do
     assert(Board.get_tile(new_board, {1, 1}) |> Tile.is_bomb?())
   end
 
-  test "returns a new board with the tile selected at the given location" do
-    old_board = Board.new().board
-
-    assert(Board.get_tile(old_board, {1, 1}) |> Tile.is_unselected?())
+  test "reveals the selected tile and all of its contiguous tiles with 0 adjacent bombs" do
```

Comment:
Just a thought: in terms of test storytelling, this test hides a bit of necessary information from the reader. To me, it reads, if we have a board of `:hidden` tiles, and we select tile at `{0, 0}`, all tiles will be reveled.

It's hard to tell that none of the `:hidden` tiles are not `{ type: :bomb}` without reading the implementation of `Board.new`, and by expecting all tiles to be revealed, it's hard to tell that we're testing that only contiguous tiles with 0 adjacent bombs are revealed.

I think your tests in `test/flood_filler_test.exs` really exercise this behavior well! I don't know that I would expect that level of detail here, so I'm wondering where the value is in this test on it's own.

I think @ecmendenhall's idea of making flood fill something that just happens when you reveal a tile will mean that `Board.select_tile(board, {0, 0})` and `FloodFiller.flood_fill(board, {0, 0})` do the same thing and that this test would become obsolete.

Reply:
@Thomascountz I definitely get what you're saying. I added this test as a kind of "I guess I should have one for this function too," but I agree that the `FloodFiller` tests cover the same behaviour and are much more expressive. I'll 🔪 it!